app/models/like.rb

```diff
@@ -0,0 +1,6 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :movie
+
+  validates :user_id, uniqueness: {scope: :movie_id}
```

Comment:
```ruby
# this will also ensure the associated user and movie are actually in the database
validates :user, uniqueness: {scope: :movie}
```
(This note applies everywhere the id of an object is validated rather than the existence of the object in the database.)